addons/misra.py

```diff
@@ -255,6 +255,36 @@ def getForLoopExpressions(forToken):
             lpar.astOperand2.astOperand2.astOperand1,
             lpar.astOperand2.astOperand2.astOperand2]
 
+def getWhileLoopExpressions(whileToken):
+    if not whileToken or whileToken.str != 'while':
+        return None
+    lpar = whileToken.next
+    if not lpar or lpar.str != '(':
+        return None
+    if not lpar.astOperand2.astOperand1:
+        return None
+    counter = lpar.astOperand2.astOperand1.str
+    token = lpar
+    while (token and token.str != "{"):
+        token = token.next
+    token_link = token
+    if token_link:
+        while (token and token != token_link.link):
+            token = token.next
+            if token.isAssignmentOp and token.astOperand1.str == counter:
+                return lpar.astOperand2
+            elif token.str == counter and token.astParent and token.astParent.str in {'++', '--'}:
+                return lpar.astOperand2
+    elif not token_link and whileToken.previous.str == '}':
```

Comment:
The `not token_link and` is redundant. You can just write:

    elif whileToken.previous.str == '}':